<!DOCTYPE html> <html lang="en"> <head> <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"> <meta charset="utf-8"> <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"> <meta http-equiv="X-UA-Compatible" content="IE=edge"> <title> Gene by environment interactions | Sahir Bhatnagar </title> <meta name="author" content="Sahir Bhatnagar"> <meta name="description" content="High-dimensional gene by environment interactions with applications to maternal and child health"> <meta name="keywords" content="jekyll, jekyll-theme, academic-website, portfolio-website"> <link rel="stylesheet" href="/assets/css/bootstrap.min.css?a4b3f509e79c54a512b890d73235ef04"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/css/mdb.min.css" integrity="sha256-jpjYvU3G3N6nrrBwXJoVEYI/0zw8htfFnhT9ljN3JJw=" crossorigin="anonymous"> <link defer rel="stylesheet" href="https://unpkg.com/bootstrap-table@1.22.1/dist/bootstrap-table.min.css"> <link rel="stylesheet" href="/assets/css/academicons.min.css?f0b7046b84e425c55f3463ac249818f5"> <link rel="stylesheet" type="text/css" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700|Roboto+Slab:100,300,400,500,700|Material+Icons"> <link rel="stylesheet" href="https://sahirbhatnagar.com/assets/css/fontawesome-all.min.css"> <link rel="stylesheet" href="https://sahirbhatnagar.com/assets/css/academicons.min.css"> <link href="https://fonts.googleapis.com/css2?family=Lora:ital,wght@0,400;0,500;0,600;0,700;1,400;1,500;1,600;1,700&amp;family=Rubik:ital,wght@0,300;0,400;0,500;0,600;0,700;1,300;1,400;1,500;1,600;1,700&amp;display=swap" rel="stylesheet"> <link href="https://fonts.googleapis.com/css2?family=Fira+Sans:wght@400;700&amp;display=swap" rel="stylesheet"> <link rel="stylesheet" href="/assets/css/jekyll-pygments-themes-github.css?591dab5a4e56573bf4ef7fd332894c99" media="" id="highlight_theme_light"> <link rel="shortcut icon" href="data:image/svg+xml,&lt;svg%20xmlns=%22http://www.w3.org/2000/svg%22%20viewBox=%220%200%20100%20100%22&gt;&lt;text%20y=%22.9em%22%20font-size=%2290%22&gt;%E2%9A%9B%EF%B8%8F&lt;/text&gt;&lt;/svg&gt;"> <link rel="stylesheet" href="/assets/css/main.css?d41d8cd98f00b204e9800998ecf8427e"> <link rel="canonical" href="https://sahirbhatnagar.com/projects/1_project/"> <link rel="stylesheet" href="/assets/css/jekyll-pygments-themes-native.css?5847e5ed4a4568527aa6cfab446049ca" media="none" id="highlight_theme_dark"> <script src="/assets/js/theme.js?bf50d6d9dd867d3e0f3b0add94449649"></script> </head> <body class="fixed-top-nav "> <header> <nav id="navbar" class="navbar navbar-light navbar-expand-sm fixed-top" role="navigation"> <div class="container"> <a class="navbar-brand title font-weight-lighter" href="/"> <span class="font-weight-bold">Sahir </span> Bhatnagar </a> <button class="navbar-toggler collapsed ml-auto" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation"> <span class="sr-only">Toggle navigation</span> <span class="icon-bar top-bar"></span> <span class="icon-bar middle-bar"></span> <span class="icon-bar bottom-bar"></span> </button> <div class="collapse navbar-collapse text-right" id="navbarNav"> <ul class="navbar-nav ml-auto flex-nowrap"> <li class="nav-item "> <a class="nav-link" href="/">about </a> </li> <li class="nav-item "> <a class="nav-link" href="/publications/">publications </a> </li> <li class="nav-item "> <a class="nav-link" href="/people/">people </a> </li> <li class="nav-item active"> <a class="nav-link" href="/projects/">projects <span class="sr-only">(current)</span> </a> </li> <li class="nav-item "> <a class="nav-link" href="/software/">software </a> </li> <li class="nav-item "> <a class="nav-link" href="/teaching/">teaching </a> </li> <li class="nav-item "> <a class="nav-link" href="/talks/">talks </a> </li> <li class="nav-item "> <a class="nav-link" href="/cv/">cv </a> </li> <li class="toggle-container"> <button id="light-toggle" title="Change theme"> <i class="fa-solid fa-moon"></i> <i class="fa-solid fa-sun"></i> </button> </li> </ul> </div> </div> </nav> <progress id="progress" value="0"> <div class="progress-container"> <span class="progress-bar"></span> </div> </progress> </header> <div class="container mt-5" role="main"> <div class="post"> <header class="post-header"> <h1 class="post-title">Gene by environment interactions</h1> <p class="post-description">High-dimensional gene by environment interactions with applications to maternal and child health</p> </header> <article> <div class="row"> <div class="col-sm mt-3 mt-md-0"> <figure> <picture> <source class="responsive-img-srcset" srcset="/assets/img/maternal-480.webp 480w,/assets/img/maternal-800.webp 800w,/assets/img/maternal-1400.webp 1400w," sizes="95vw" type="image/webp"></source> <img src="/assets/img/maternal.png" class="img-fluid rounded z-depth-1" width="100%" height="auto" title="example image" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> </div> <div class="caption"> Figure 1. Motivating dataset: Children born to mothers with gestational diabetes mellitus (GDM) are more likely to develop obesity or become overweight. Epigenetic factors, such as DNA methylation, are suspected to play a significant role in mediating the effects of GDM on childhood obesity. </div> <p>High-dimensional (HD) genomic data \((\mathbf{X})\) often reflect a tissue’s functioning and can be strongly influenced by lifestyle or demographic exposures \((E)\), making the interactions \((\mathbf{X} \cdot E)\) crucial for predicting outcomes \((Y)\). However, the sheer number of potential interactions, their non-linear nature, and the low power to detect them pose major challenges. In precision medicine, where the goal is to tailor interventions to individual risk profiles, it is equally important to build accurate models and to uncover which features—among possibly correlated or latent factors—actually drive predictions. Recent evidence also suggests that environmental exposures can broadly alter regulatory networks and that assessing gene similarity or co-regulation can capture these systemic effects better than simply modeling raw expression data. Yet most existing predictive methods that use HD data and exposures struggle with low power, limited interpretability, and difficulty handling interactions, underscoring the need for more appropriate analytic approaches.</p> <p>To address this gap, we propose ECLUST <a class="citation" href="#bhatnagar2018analytic">(Bhatnagar et al., 2018)</a>, a conceptual analytic strategy that uses exposure-sensitive data clusters in two-step algorithms for continuous or binary outcome prediction in HD contexts. We hypothesize that incorporating exposure data into variable grouping can improve both predictive accuracy and interpretability. Our approach is motivated by maternal-child health research, where events during pregnancy—such as gestational diabetes mellitus (GDM)—are suspected to influence the risk of childhood obesity. Indeed, children born to women with GDM are more likely to be overweight or obese, and epigenetic factors have emerged as crucial mediators. Methylation changes in placenta and cord blood have been linked to GDM, and here we explore how these changes relate to childhood obesity outcomes at around five years of age. By grouping features that exhibit a systematic response to GDM, ECLUST retains a systems-based perspective and avoids focusing solely on single markers. Through simulations and real data analyses, this approach has demonstrated improved prediction accuracy and interpretability in both linear and non-linear settings, shedding light on how exposures like GDM reshape genomic co-regulation and ultimately influence childhood obesity risk. An <a href="https://cran.r-project.org/package=eclust" target="_blank" rel="external nofollow noopener">R package implementing ECLUST is available on CRAN</a>, making it accessible to a wide range of researchers and practitioners.</p> <p>Building on this foundation of exposure-centric modeling, we developped a new penalized regression method called SAIL that specifically targets non-linear interactions between a single key exposure and a large set of features under strong or weak heredity constraints <a class="citation" href="#bhatnagar2023sparse">(Bhatnagar et al., 2023)</a>. This means that an interaction term is included in the model only if the corresponding main effects are also present (strong) or at least one of them is present (weak). Importantly, SAIL is shown to possess the oracle property asymptotically, implying that it can recover the true underlying model as well as if that model were known in advance. From a practical standpoint, SAIL employs a computationally efficient fitting algorithm with automatic selection of tuning parameters, allowing it to scale well to modern HD datasets. Simulation studies demonstrate its superiority over other penalized methods in accurately predicting outcomes and recovering the true set of active (and interacting) variables when interactions are genuinely non-linear. The proposed algorithms are implemented in an <a href="https://github.com/sahirbhatnagar/sail" target="_blank" rel="external nofollow noopener">R package available on GitHub</a>.</p> <p>A compelling example of SAIL’s utility comes from a study of the Nurse Family Partnership (NFP) program. NFP is a psychosocial intervention program targeting low-income mothers with the goal of improving pregnancy outcomes, children’s health and development, and long-term economic self-sufficiency. Beginning in pregnancy and continuing through infancy, mothers receive regular home visits from nurses who provide guidance on maternal health, parenting, and mother-infant interactions. In a randomized trial, pregnant women were assigned either to the nurse-visited group (intervention) or a control group, and child IQ (using Stanford Binet scores) was measured at four years of age. In this scenario, researchers were interested in how a child’s genetic risk for educational attainment—summarized by a polygenic risk score (PRS)—might interact with participation in NFP to influence IQ at age four. Applying SAIL revealed that children with lower PRS scores (i.e., those genetically predisposed to lower educational attainment) benefited most from the intervention, indicating a clear gene-environment interaction (see Figure 2).</p> <div class="row"> <div class="col-sm mt-3 mt-md-0"> <figure> <picture> <source class="responsive-img-srcset" srcset="/assets/img/sail_nfp-480.webp 480w,/assets/img/sail_nfp-800.webp 800w,/assets/img/sail_nfp-1400.webp 1400w," sizes="95vw" type="image/webp"></source> <img src="/assets/img/sail_nfp.png" class="img-fluid rounded z-depth-1" width="100%" height="auto" title="example image" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> </div> <div class="caption"> Figure 2. Estimated interaction effect identified by the weak heredity sail using cubic B-splines and α = 0.1 for the Nurse Family Partnership data. The selected model, chosen via 10-fold cross-validation, contained three variables: the main effects for the intervention and the PRS for educational attainment using genetic variants significant at the 0.0001 level, as well as their interaction. </div> <p>A third advancement expands hierarchical gene-environment interaction (GEI) modeling to high-dimensional mixed models. A longstanding challenge in GEI analysis is that population structure and closer relatedness, alongside shared environmental exposures, can create correlations that lead to spurious signals if not properly modeled. Recognizing that ignoring these dependencies can inflate false positive rates under polygenic models, my PhD student, Julien, developed a penalized quasi-likelihood approach for hierarchical variable selection within generalized linear mixed models (GLMMs) called pglmm <a class="citation" href="#st2023hierarchical">(St-Pierre et al., 2024)</a>. By incorporating random effects to account for both genetic and environmental correlations, pglmm reduces false discoveries for main and interaction effects and achieves higher F1 scores than existing methods in simulation studies. The method is implemented in an <a href="https://github.com/julstpierre/PenalizedGLMM" target="_blank" rel="external nofollow noopener">open source Julia package</a>.</p> <p>A motivating application of pglmm centers on the Orofacial Pain: Prospective Evaluation and Risk Assessment (OPPERA) study, which previously reported significant associations between temporomandibular disorder (TMD)—a painful jaw condition disproportionately affecting females—and four distinct genetic loci. Because TMD may involve sex-specific pathophysiological mechanisms, pglmm was used to identify important sex-specific predictors in the OPPERA discovery cohort and then evaluate the model’s predictive performance in two replication cohorts: OPPERA II Chronic TMD Replication and Complex Persistent Pain Conditions (CPPC). pglmm successfully retrieved the previously implicated loci, highlighting its ability to detect biologically relevant interactions and confirm risk predictions in independent samples. These results underscore the importance of accounting for correlated structures in GEI analyses and demonstrate how hierarchical selection in a mixed-model framework can improve both discovery and replication in complex trait genetics.</p> </article> <h2>References</h2> <div class="publications"> <h2 class="bibliography">2024</h2> <ol class="bibliography"><li><div class="row"> <div class="col col-sm-2 abbr"> <figure> <picture> <source class="responsive-img-srcset" srcset="/assets/img/publication_preview/smma.webp-480.webp 480w,/assets/img/publication_preview/smma.webp-800.webp 800w,/assets/img/publication_preview/smma.webp-1400.webp 1400w," sizes="200px" type="image/webp"> <img src="/assets/img/publication_preview/smma.webp" class="preview z-depth-1 rounded" width="100%" height="auto" alt="smma.webp" data-zoomable onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </source></picture> </figure> </div> <div id="st2023hierarchical" class="col-sm-8"> <div class="title">Hierarchical selection of genetic and gene by environment interaction effects in high-dimensional mixed models</div> <div class="author"> Julien St-Pierre, Karim Oualkacha, and Sahir Rai Bhatnagar </div> <div class="periodical"> <em>Statistical Methods in Medical Research</em>, 2024 </div> <div class="periodical"> </div> <div class="links"> <a href="https://doi.org/10.1177/09622802241293768" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">DOI</a> <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a> <a href="/assets/pdf/smmr_2024.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> <a href="https://github.com/julstpierre/PenalizedGLMM" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">Code</a> </div> <div class="badges"> <span class="altmetric-embed" data-badge-type="2" data-badge-popover="right" data-doi="10.1177/09622802241293768"></span> <a href="https://scholar.google.com/citations?view_op=view_citation&amp;hl=en&amp;user=9XJum2AAAAAJ&amp;citation_for_view=9XJum2AAAAAJ:SP6oXDckpogC" aria-label="Google Scholar link" role="button" rel="external nofollow noopener" target="_blank"> <img src="https://img.shields.io/badge/scholar-0-4285F4?logo=googlescholar&amp;labelColor=beige" alt="0 Google Scholar citations"> </a> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">st2023hierarchical</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Hierarchical selection of genetic and gene by environment interaction effects in high-dimensional mixed models}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{St-Pierre, Julien and Oualkacha, Karim and Rai Bhatnagar, Sahir}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{Statistical Methods in Medical Research}</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">{09622802241293768}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2024}</span><span class="p">,</span>
  <span class="na">publisher</span> <span class="p">=</span> <span class="s">{SAGE Publications Sage UK: London, England}</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">{10.1177/09622802241293768}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div></li></ol> <h2 class="bibliography">2023</h2> <ol class="bibliography"><li><div class="row"> <div class="col col-sm-2 abbr"> <figure> <picture> <source class="responsive-img-srcset" srcset="/assets/img/publication_preview/csda_preview-480.webp 480w,/assets/img/publication_preview/csda_preview-800.webp 800w,/assets/img/publication_preview/csda_preview-1400.webp 1400w," sizes="200px" type="image/webp"> <img src="/assets/img/publication_preview/csda_preview.jpg" class="preview z-depth-1 rounded" width="100%" height="auto" alt="csda_preview.jpg" data-zoomable onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </source></picture> </figure> </div> <div id="bhatnagar2023sparse" class="col-sm-8"> <div class="title">A sparse additive model for high-dimensional interactions with an exposure variable</div> <div class="author"> <em>Sahir R Bhatnagar</em>, Tianyuan Lu, Amanda Lovato, David L Olds, Michael S Kobor, Michael J Meaney, Kieran O’Donnell, Archer Y Yang, and Celia MT Greenwood </div> <div class="periodical"> <em>Computational Statistics &amp; Data Analysis</em>, 2023 </div> <div class="periodical"> </div> <div class="links"> <a href="https://doi.org/10.1016/j.csda.2022.107624" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">DOI</a> <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a> <a href="/assets/pdf/csda_2023.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> <a href="https://sahirbhatnagar.com/sail/" class="btn btn-sm z-depth-0" role="button">Code</a> </div> <div class="badges"> <span class="altmetric-embed" data-badge-type="2" data-badge-popover="right" data-doi="10.1016/j.csda.2022.107624"></span> <a href="https://scholar.google.com/citations?view_op=view_citation&amp;hl=en&amp;user=9XJum2AAAAAJ&amp;citation_for_view=9XJum2AAAAAJ:LkGwnXOMwfcC" aria-label="Google Scholar link" role="button" rel="external nofollow noopener" target="_blank"> <img src="https://img.shields.io/badge/scholar-10-4285F4?logo=googlescholar&amp;labelColor=beige" alt="10 Google Scholar citations"> </a> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">bhatnagar2023sparse</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{A sparse additive model for high-dimensional interactions with an exposure variable}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Bhatnagar, Sahir R and Lu, Tianyuan and Lovato, Amanda and Olds, David L and Kobor, Michael S and Meaney, Michael J and O'Donnell, Kieran and Yang, Archer Y and Greenwood, Celia MT}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{Computational Statistics \&amp; Data Analysis}</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">{179}</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">{107624}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2023}</span><span class="p">,</span>
  <span class="na">publisher</span> <span class="p">=</span> <span class="s">{North-Holland}</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">{10.1016/j.csda.2022.107624}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div></li></ol> <h2 class="bibliography">2018</h2> <ol class="bibliography"><li><div class="row"> <div class="col col-sm-2 abbr"> <figure> <picture> <source class="responsive-img-srcset" srcset="/assets/img/publication_preview/gepi-480.webp 480w,/assets/img/publication_preview/gepi-800.webp 800w,/assets/img/publication_preview/gepi-1400.webp 1400w," sizes="200px" type="image/webp"> <img src="/assets/img/publication_preview/gepi.jpg" class="preview z-depth-1 rounded" width="100%" height="auto" alt="gepi.jpg" data-zoomable onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </source></picture> </figure> </div> <div id="bhatnagar2018analytic" class="col-sm-8"> <div class="title">An analytic approach for interpretable predictive models in high-dimensional data in the presence of interactions with exposures</div> <div class="author"> <em>Sahir Rai Bhatnagar</em>, Yi Yang, Budhachandra Khundrakpam, Alan C Evans, Mathieu Blanchette, Luigi Bouchard, and Celia MT Greenwood </div> <div class="periodical"> <em>Genetic epidemiology</em>, 2018 </div> <div class="periodical"> </div> <div class="links"> <a href="https://doi.org/10.1002/gepi.22112" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">DOI</a> <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a> <a href="/assets/pdf/eclust_paper.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> <a href="https://sahirbhatnagar.com/eclust/" class="btn btn-sm z-depth-0" role="button">Code</a> </div> <div class="badges"> <span class="altmetric-embed" data-badge-type="2" data-badge-popover="right" data-doi="10.1002/gepi.22112"></span> <a href="https://scholar.google.com/citations?view_op=view_citation&amp;hl=en&amp;user=9XJum2AAAAAJ&amp;citation_for_view=9XJum2AAAAAJ:Tyk-4Ss8FVUC" aria-label="Google Scholar link" role="button" rel="external nofollow noopener" target="_blank"> <img src="https://img.shields.io/badge/scholar-7-4285F4?logo=googlescholar&amp;labelColor=beige" alt="7 Google Scholar citations"> </a> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">bhatnagar2018analytic</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{An analytic approach for interpretable predictive models in high-dimensional data in the presence of interactions with exposures}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Bhatnagar, Sahir Rai and Yang, Yi and Khundrakpam, Budhachandra and Evans, Alan C and Blanchette, Mathieu and Bouchard, Luigi and Greenwood, Celia MT}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{Genetic epidemiology}</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">{42}</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">{3}</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">{233--249}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2018}</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">{10.1002/gepi.22112}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div></li></ol> </div> </div> </div> <footer class="fixed-bottom" role="contentinfo"> <div class="container mt-0"> © Copyright 2025 Sahir Bhatnagar. Powered by <a href="http://jekyllrb.com/" target="_blank" rel="external nofollow noopener">Jekyll</a>, <a href="https://github.com/alshedivat/al-folio" rel="external nofollow noopener" target="_blank">al-folio</a>, <a href="http://daringfireball.net/projects/markdown/" rel="external nofollow noopener" target="_blank">markdown</a>, <a href="https://cran.r-project.org/" rel="external nofollow noopener" target="_blank">R</a>, <a href="http://yihui.name/knitr/" rel="external nofollow noopener" target="_blank">knitr</a>, <a href="http://johnmacfarlane.net/pandoc/" rel="external nofollow noopener" target="_blank">pandoc</a>, <a href="https://github.com/mozilla/Fira" rel="external nofollow noopener" target="_blank">Fira</a> and <a href="https://github.com/sahirbhatnagar/sahirbhatnagar.github.io" rel="external nofollow noopener" target="_blank">GitHub</a> </div> </footer> <script src="https://cdn.jsdelivr.net/npm/jquery@3.6.0/dist/jquery.min.js" integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script> <script src="/assets/js/bootstrap.bundle.min.js"></script> <script src="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/js/mdb.min.js" integrity="sha256-NdbiivsvWt7VYCt6hYNT3h/th9vSTL4EDWeGs5SN3DA=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/masonry-layout@4.2.2/dist/masonry.pkgd.min.js" integrity="sha256-Nn1q/fx0H7SNLZMQ5Hw5JLaTRZp0yILA/FRexe19VdI=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/imagesloaded@4/imagesloaded.pkgd.min.js"></script> <script defer src="/assets/js/masonry.js" type="text/javascript"></script> <script type="text/javascript">$(function(){$('[data-toggle="tooltip"]').tooltip()});</script> <script defer src="https://cdn.jsdelivr.net/npm/medium-zoom@1.0.8/dist/medium-zoom.min.js" integrity="sha256-7PhEpEWEW0XXQ0k6kQrPKwuoIomz8R8IYyuU1Qew4P8=" crossorigin="anonymous"></script> <script defer src="/assets/js/zoom.js?85ddb88934d28b74e78031fd54cf8308"></script> <script defer src="https://unpkg.com/bootstrap-table@1.22.1/dist/bootstrap-table.min.js"></script> <script src="/assets/js/no_defer.js?2930004b8d7fcd0a8e00fdcfc8fc9f24"></script> <script defer src="/assets/js/common.js?4a129fbf39254905f505c7246e641eaf"></script> <script defer src="/assets/js/copy_code.js?8664456308d8a0b76907c75d01dd1dbf" type="text/javascript"></script> <script async src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script> <script async src="https://badge.dimensions.ai/badge.js"></script> <script type="text/javascript">window.MathJax={tex:{tags:"ams"}};</script> <script defer type="text/javascript" id="MathJax-script" src="https://cdn.jsdelivr.net/npm/mathjax@3.2.0/es5/tex-mml-chtml.js"></script> <script defer src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script> <script async src="https://www.googletagmanager.com/gtag/js?id=UA-60189402-1"></script> <script>function gtag(){window.dataLayer.push(arguments)}window.dataLayer=window.dataLayer||[],gtag("js",new Date),gtag("config","UA-60189402-1");</script> <script type="text/javascript">function progressBarSetup(){"max"in document.createElement("progress")?(initializeProgressElement(),$(document).on("scroll",function(){progressBar.attr({value:getCurrentScrollPosition()})}),$(window).on("resize",initializeProgressElement)):(resizeProgressBar(),$(document).on("scroll",resizeProgressBar),$(window).on("resize",resizeProgressBar))}function getCurrentScrollPosition(){return $(window).scrollTop()}function initializeProgressElement(){let e=$("#navbar").outerHeight(!0);$("body").css({"padding-top":e}),$("progress-container").css({"padding-top":e}),progressBar.css({top:e}),progressBar.attr({max:getDistanceToScroll(),value:getCurrentScrollPosition()})}function getDistanceToScroll(){return $(document).height()-$(window).height()}function resizeProgressBar(){progressBar.css({width:getWidthPercentage()+"%"})}function getWidthPercentage(){return getCurrentScrollPosition()/getDistanceToScroll()*100}const progressBar=$("#progress");window.onload=function(){setTimeout(progressBarSetup,50)};</script> </body> </html>